### ISSUE TITLE
Extremely slow computation of band depths

### ISSUE THREAD

**Issue by dmey** [statsmodels/statsmodels#7058]
I am using [`statsmodels.graphics.functional.fboxplot`](https://www.statsmodels.org/stable/generated/statsmodels.graphics.functional.fboxplot.html) to compute band depths for a dataset comprising of 10k 200-time-point curves however the computation seems extremely slow (> day). I noticed that the implementation described in the doc strings references that of [Sun et al., 2012](https://doi.org/10.1002/sta4.8) in which he evaluated one million curves, at fifty time points each, in 12.4 seconds with Matlab. I wonder if anyone has bench-marked the function on a similar dataset or this may be a problem with the data I am working with. During the fitting I only get the following warning which may be of relevance --

```
/usr/local/miniconda3/envs/statsmodels/lib/python3.8/site-packages/statsmodels/graphics/functional.py:848: RuntimeWarning: invalid value encountered in double_scalars
  normfactor = factorial(num) / 2. / factorial(num - 2)
```
Running on a 48 Xeon CPU machine with 128 GB RAM, Ubuntu 18.04, Python 3.8.5, Statsmodels 0.11.1.

**Comment by josef-pkt** [comment]
there was a change in code since 0.11 to speed this up.
#6744

try updating to statsmodels 0.12

**Comment by dmey** [comment]
@josef-pkt light-speed! Can confirm that this was indeed solved in #6744. Sorry for missing this!

**Comment by josef-pkt** [comment]
I only found it because your warning line `statsmodels/graphics/functional.py:848` doesn't exist in current code.
